src/GraphQL/Conversion/IListConverter.cs

```diff
@@ -0,0 +1,17 @@
+namespace GraphQL.Conversion;
+
+/// <summary>
+/// A converter that can convert a list of objects to a specific list type, such as <see cref="HashSet{T}"/>.
+/// </summary>
+public interface IListConverter
+{
+    /// <summary>
+    /// The type of the list element.
+    /// </summary>
+    Type ElementType { get; }
+
+    /// <summary>
+    /// Converts a list of objects to a specific list type, such as <see cref="HashSet{T}"/>.
+    /// </summary>
+    object Convert(object?[] list);
+}
```

Comment:
The list converter itself primarily just has a `Convert` method to convert a list of objects (`object[]`) into the intended instance; in my example, to `HashSet<int>`.